src/Microsoft.IdentityModel.Xml/EnvelopedSignatureReader.cs

```diff
@@ -99,11 +99,9 @@ public override bool Read()
             bool result = InnerReader.Read();
             if (result
                 && InnerReader.IsLocalName(XmlSignatureConstants.Elements.Signature)
-                && InnerReader.IsNamespaceUri(XmlSignatureConstants.Namespace))
+                && InnerReader.IsNamespaceUri(XmlSignatureConstants.Namespace)
+                && Signature == null)
```

Comment:
This will only recognize the first signature, correct?

Reply:
Yes, it will assume that the first signature is the right one because in SAML2 that is always the case. The SAML2 XML schema requires the signature to be before any place where a nested signature can occur for all types of messages/data. I checked the spec when I wrote this PR, can get back with proper references to the spec if you want to have that.